core/src/main/scala/org/locationtech/rasterframes/expressions/focalops/SurfaceOp.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * This software is licensed under the Apache 2 license, quoted below.
+ *
+ * Copyright 2021 Azavea, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *     [http://www.apache.org/licenses/LICENSE-2.0]
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.locationtech.rasterframes.expressions.focalops
+
+import org.slf4j.LoggerFactory
+import com.typesafe.scalalogging.Logger
+import org.apache.spark.sql.types.DataType
+import org.locationtech.rasterframes.expressions.row
+import org.apache.spark.sql.catalyst.expressions.codegen.CodegenFallback
+import org.locationtech.rasterframes.ref.RasterRef
+
+import org.locationtech.rasterframes.encoders.syntax._
+import org.locationtech.rasterframes.expressions.DynamicExtractors._
+import geotrellis.raster.Tile
+import org.apache.spark.sql.catalyst.analysis.TypeCheckResult
+import org.apache.spark.sql.catalyst.analysis.TypeCheckResult.{TypeCheckFailure, TypeCheckSuccess}
+import org.apache.spark.sql.catalyst.expressions.UnaryExpression
+import org.locationtech.rasterframes.model.TileContext
+import org.locationtech.rasterframes.expressions.NullToValue
+import org.locationtech.rasterframes.tiles.ProjectedRasterTile
+
+/** Operation on a tile returning a tile. */
+trait SurfaceOp extends UnaryExpression with NullToValue with CodegenFallback  {
```

Comment:
How is this different from [`UnaryRasterOp`](https://github.com/locationtech/rasterframes/blob/develop/core/src/main/scala/org/locationtech/rasterframes/expressions/UnaryRasterOp.scala)

Reply:
@metasim _have no idea_; I also haven't renamed classes yet (talking about your changes in your PR).

Reply:
@metasim restructured the hierarchy a bit and aligned with your changes from other PRs. I hope it looks a bit better now.